cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/PluginInstantiator.java

```diff
@@ -211,8 +266,10 @@ public void close() throws IOException {
     @Override
     public ClassLoader load(ArtifactDescriptor key) throws Exception {
       File unpackedDir = DirUtils.createTempDir(tmpDir);
-      BundleJarUtil.unpackProgramJar(key.getLocation(), unpackedDir);
-
+      String artifactFile = String.format("%s/artifacts/%s/%s", key.getArtifact().getNamespace().getId(),
```

Comment:
ok wasn't clear at first that this is meant to be the same path as the actual file for local mode.  This is brittle and will break in a confusing way if somebody changes the implementation of the write method in ArtifactStore.  They should share the same code.